src/crypto/CHIPCryptoPAL.h

```diff
@@ -286,6 +292,17 @@ class ECPKeypair
      **/
     virtual CHIP_ERROR ECDSA_sign_hash(const uint8_t * hash, size_t hash_length, Sig & out_signature) = 0;
 
+    /**
+     * @brief A function to sign an attestation data blob using ECDSA
+     * @param attestationElements Span of the AttestationElements to be signed
+     * @param attestationChallenge Span of the AttestationChallenge to be signed
+     * @param out_signature Buffer that will hold the output signature. The signature consists of: 2 EC elements (r and s),
+     * in raw <r,s> point form (see SEC1).
+     * @return Returns a CHIP_ERROR on error, CHIP_NO_ERROR otherwise
+     **/
+    virtual CHIP_ERROR ECDSA_sign_attestation_data(const ByteSpan & attestationElements, const ByteSpan & attestationChallenge,
+                                                   Sig & out_signature) = 0;
```

Comment:
This does not belong in ECPKeyPair public API. Attestation signature is always P256 and this special casing with the attestation challenge should just be done by *using* a keypair, rather than being *part of* keypair.